docs/shallow_water_equation.md

```diff
@@ -0,0 +1,194 @@
+
+
+## Governing equations
+
+The shallow water equations are given by
+
+\begin{equation}
+\label{eqn:swe}
+\frac{\partial\mathbf{U}}{\partial t} + \frac{\partial \mathbf{E}}{\partial x} + \frac{\partial \mathbf{G}}{\partial x} = \mathbf{S}
+\end{equation}
+
+where 
+
+\begin{align}
+  \mathbf{U}
+  =
+  \begin{bmatrix}
+  h \\[.5em]
+  hu \\[.5em]
+  hv
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{E}
+  =
+  \begin{bmatrix}
+  hu \\[.5em]
+  hu^2 + \frac{1}{2}gh^2 \\[.5em]
+  huv
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{G}
+  =
+  \begin{bmatrix}
+  hv \\[.5em]
+  huv \\[.5em]
+  hv^2 + \frac{1}{2}gh^2
+  \end{bmatrix}
+\end{align}
+
+\begin{align}
+  \mathbf{S}
+  =
+  \begin{bmatrix}
+  Q_i \\[.5em]
+  -gh\frac{\partial z}{\partial x} - C_D u \sqrt{u^2 + v^2}\\[.5em]
+  -gh\frac{\partial z}{\partial y} - C_D v \sqrt{u^2 + v^2}
+  \end{bmatrix}
+\end{align}
+
+$h$ is the flow depth,
+$u$ is the vertically-averaged velocity in x-direction,
+$v$ is the vertically-averaged velocity in x-direction,
+$z$ is the bed elevation,
+$C_D = g n^2/h^\frac{1}{3}$ is the drag coefficient, and
+$n$ is the Manning's coefficient.
+
+## Spatial discretization
+
+\begin{eqnarray}
+\frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\int_\Omega \frac{\partial\mathbf{E}}{\partial x}  d\Omega + 
+\int_\Omega \frac{\partial\mathbf{G}}{\partial x}  d\Omega +  &=&
+\int_\Omega \mathbf{S} d\Omega \nonumber\\
+\frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\oint_{d\Omega} \left( \mathbf{E}dy  - \mathbf{G} dx \right) ds &=&
+\int_\Omega \mathbf{S} d\Omega \nonumber\\
+ \frac{\partial}{\partial t} \int_\Omega \mathbf{U} d\Omega + 
+\int_{d\Omega} \left( \mathbf{F} . \mathbf{n} \right) ds +  &=&
+\int_\Omega \mathbf{S} d\Omega
+\end{eqnarray}
+
+where
+
+$\mathbf{F}$ is the flux vector and
+$\mathbf{n}$ is the outward pointing unit vector to the boundary $\partial\Omega$. 
+The flux normal flux across the face, $\mathbf{F.n}$, is given by
+
+\begin{align}
+  \mathbf{F.n}
+  =
+  \begin{bmatrix}
+  hu_\perp  \\[.5em]
+  huu_\perp + \frac{1}{2}gh^2 \cos\phi \\[.5em]
+  hvu_\perp + \frac{1}{2}gh^2 \sin\phi 
```

Comment:
+ /dfrac{1}{24}g(\Delta h)^{2} \sin \phi

Reply:
This third term is in Kim et al. (2012) but isn't in the series of papers by Sanders. I can't figure out which paper derived this third term. Do you know?

Reply:
Okay, I figure out the origin of the third term. It is eq 13 in Bradford and Sanders (2002) paper (https://ascelibrary.org/doi/10.1061/%28ASCE%290733-9429%282002%29128%3A3%28289%29)